pkg/node/discovery/drive_matcher.go

```diff
@@ -100,7 +98,7 @@ func (d *Discovery) identifyDriveByLegacyName(localDriveState directcsi.DirectCS
 	}
 
 	for i, remoteDrive := range d.remoteDrives {
-		if !remoteDrive.matched && remoteDrive.Name == v1beta1DriveName {
+		if !remoteDrive.matched && remoteDrive.Status.NodeName == d.NodeID && remoteDrive.Status.Path == localDriveState.Path {
```

Comment:
We don't need this condition - `remoteDrive.Status.NodeName == d.NodeID` as the `remoteDrives` are already filtered by NodeName

Reply:
Done